test/quantization/core/test_quantized_module.py

```diff
@@ -889,6 +889,187 @@ def test_embedding_bag_api(self, num_embeddings, embedding_dim, num_offsets, set
+
+        X_fp32 = torch.randn(*([in_channels] * dim))
+        s, z = _calculate_dynamic_qparams(X_fp32, dtype, reduce_range)
+        X_q = torch.quantize_per_tensor(X_fp32, s, z, dtype)
+        X_dq = torch.dequantize(X_q)
+
+        quantized_module = q_mod(in_channels, out_channels, kernel_size, stride=stride, padding=padding,
+                                 dilation=dilation, groups=groups, bias=bias, padding_mode=padding_mode)
+        dynamic_module = dq_mod(in_channels, out_channels, kernel_size, stride=stride, padding=padding,
+                                dilation=dilation, groups=groups, bias=bias, padding_mode=padding_mode)
+
+        quantized_module.scale, quantized_module.zero_point = s, z
+        dynamic_module.set_weight_bias(*quantized_module._weight_bias())
+
+        Y_q_ref = quantized_module(X_q)
+        Y_ref = torch.dequantize(Y_q_ref)
+
+        Y = dynamic_module(X_dq, reduce_range)
+
+        self.assertEqual(Y, Y_ref)
+
+        # Test serialization of quantized Conv Module using state_dict
+        W_q, b = dynamic_module._weight_bias()
+        model_dict = dynamic_module.state_dict()
+        self.assertEqual(model_dict['weight'], W_q)
+        self.assertEqual(model_dict['bias'], b)
+        bytes_io = io.BytesIO()
+        torch.save(model_dict, bytes_io)
+        bytes_io.seek(0)
+        loaded_dict = torch.load(bytes_io)
+        for key in loaded_dict:
+            self.assertEqual(model_dict[key], loaded_dict[key])
+        loaded_qconv_module = type(dynamic_module)(
+            in_channels, out_channels, kernel_size, stride=stride, padding=padding,
+            dilation=dilation, groups=groups, bias=bias, padding_mode=padding_mode)
+        loaded_qconv_module.load_state_dict(loaded_dict)
+
+        self.assertTrue(dir(loaded_qconv_module) == dir(dynamic_module))
+        self.assertTrue(dynamic_module._get_name() == loaded_qconv_module._get_name())
+        self.assertTrue(hasattr(loaded_qconv_module, '_packed_params'))
+        self.assertTrue(hasattr(loaded_qconv_module, '_weight_bias'))
+
+        self.assertEqual(dynamic_module.weight(), loaded_qconv_module.weight())
+        if bias:
+            self.assertEqual(dynamic_module.bias(), loaded_qconv_module.bias())
+        self.assertEqual(dynamic_module.scale, loaded_qconv_module.scale)
+        self.assertEqual(dynamic_module.zero_point,
+                         loaded_qconv_module.zero_point)
+        Y_loaded = loaded_qconv_module(X_fp32, reduce_range)
+        np.testing.assert_array_almost_equal(
+            Y.numpy(), Y_loaded.numpy(), decimal=0)
+
+        # Test serialization
+        b = io.BytesIO()
+        torch.save(dynamic_module, b)
+        b.seek(0)
+        loaded_conv = torch.load(b)
+
+        self.assertEqual(loaded_conv.bias(), dynamic_module.bias())
+        self.assertEqual(loaded_conv.scale, dynamic_module.scale)
+        self.assertEqual(loaded_conv.zero_point,
+                         dynamic_module.zero_point)
```

Comment:
Didnt the lines above already test serialization?

Reply:
I'm just copying from the qlinear_dynamic test, not 100% sure what the specific corner case its looking to check

Reply:
ok I can just fix